### PR TITLE
Fixed import and visibility tax values to lowercase

### PR DESCRIPTION
**Pull request by ajayghaghretiya-multidots**
### All Submissions:

-   [X] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [X] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #[33794](https://github.com/woocommerce/woocommerce/issues/33794).

### How to test the changes in this Pull Request:

1. Add the code changes to your local site.
2. Check the import tool again with keeping visibility and tax value as "Visible", and "Taxable".
3. Product will be imported successfully as per the below screenshot.

CleanShot 2022-07-12 at 13 31 23

### Other information:

-   [X] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [X] Have you successfully run tests with your changes locally?
-   [X] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/includes/abstracts/abstract-wc-product.php [hunk 1]
```diff
@@ -782,6 +782,10 @@ public function set_featured( $featured ) {
 	 */
 	public function set_catalog_visibility( $visibility ) {
 		$options = array_keys( wc_get_product_visibility_options() );
+
+		//Convert to lowercase
+		$visibility = strtolower( $visibility );
```
Comment (jorgeatorres) [inline plugins/woocommerce/includes/abstracts/abstract-wc-product.php]: Here we need to be a bit more careful: given `wc_get_product_visibility_options()` calls a filter, plugins can add their own visibility options, and we can't be sure those correspond to lowercase keys, like the default ones. That is, the options returned by `wc_get_product_visibility_options()` might be case sensitive.

As such, I think what we want to do here is first check if `$visibility` is in `$options`. If it is, we use it verbatim. If it isn't, _but_ `strtolower( $visibility )` is, we use the lowercase version.

That would mean we respect case sensitivity but at the same time makes things easier, especially for the default visibility options.

**Review by jorgeatorres — CHANGES_REQUESTED** [review]
Hi @ajayghaghretiya-multidots!

First of all, let me apologize for the delay in getting this reviewed and thank you for your contribution! 💯
Overall, the changes look good. I left one tiny request, but once addressed we should be good to go.

I understand it's been a while since you initially submitted this PR, so please let us know if you have the time to continue working on it or if you'd rather have us take over.

Looking forward to your response!

File: plugins/woocommerce/includes/abstracts/abstract-wc-product.php (hunk 1 above)
Comment (ajayghaghretiya-multidots) [reply plugins/woocommerce/includes/abstracts/abstract-wc-product.php]: @jorgeatorres No problem with the delay. I understand WooCommerce has many other priorities going on. Here, Can we first check the visibility options with the original case? If isn't a match then we can do a check-in with strtolower? I would be happy to help with this ticket.

WDYT?

Comment (jorgeatorres) [reply plugins/woocommerce/includes/abstracts/abstract-wc-product.php]: Hi @ajayghaghretiya-multidots!

Thanks for your understanding. The approach you've suggested sounds reasonable.
Feel free to push the relevant changes and ask for a re-review from me. Thanks again!

**Comment by samueljseay** [comment]
We are closing this pull request, as the last activity has been quite some time ago. If you are still interested or you feel like this is a mistake, please open another PR based on a recent version of the trunk branch. Thanks!

**Comment by jorgeatorres** [comment]
@rrennick: I wanted to rescue this PR (authored by a community contributor) from the "purge". IMO it only needed a tiny change (see https://github.com/woocommerce/woocommerce/pull/33820#pullrequestreview-1217956903), so I went head and added it on their behalf. Please let me know what you think.

**Review by rrennick — APPROVED** [review]
LGTM

**Comment by rrennick** [comment]
@jorgeatorres I think you need to approve this before merging. It looks good so feel free to merge.

**Review by jorgeatorres — APPROVED** [review]

**Comment by jorgeatorres** [comment]
Thank you @rrennick!

**Pull request merged**